src/main/java/seedu/address/MainApp.java

```diff
@@ -54,46 +62,33 @@ public void init() throws Exception {
 
+        //Initialise UserPrefs
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         UserPrefs userPrefs = initPrefs(userPrefsStorage);
-        //TODO: Update this with the 4 different EntityLists
-        AddressBookStorage addressBookStorage = new JsonAddressBookStorage(userPrefs.getAddressBookFilePath());
-        storage = new StorageManager(addressBookStorage, userPrefsStorage);
 
-        initLogging(config);
 
-        model = initModelManager(storage, userPrefs);
+        //Initialise EntityList Storage and AlfredStorage
+        ParticipantListStorage participantListStorage = new JsonParticipantListStorage(userPrefs.getParticipantListFilePath());
+        MentorListStorage mentorListStorage = new JsonMentorListStorage(userPrefs.getMentorListFilePath());
+        TeamListStorage teamListStorage = new JsonTeamListStorage(userPrefs.getTeamListFilePath();
+         alfredStorage = new AlfredStorageManager(participantListStorage, mentorListStorage, teamListStorage, userPrefsStorage);
 
-        logic = new LogicManager(model, storage);
+         initLogging(config);
+
+        model = initModelManager(alfredStorage, userPrefs);
+
+        logic = new LogicManager(model, alfredStorage);
 
         ui = new UiManager(logic);
     }
 
-    /**
-     * Returns a {@code ModelManager} with the data from {@code storage}'s address book and {@code userPrefs}. <br>
-     * The data from the sample address book will be used instead if {@code storage}'s address book is not found,
-     * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
-     */
-    private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
-        Optional<ReadOnlyAddressBook> addressBookOptional;
-        ReadOnlyAddressBook initialData;
-        try {
-            addressBookOptional = storage.readAddressBook();
-            if (!addressBookOptional.isPresent()) {
-                logger.info("Data file not found. Will be starting with a sample AddressBook");
-            }
-            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
-        } catch (DataConversionException e) {
-            logger.warning("Data file not in the correct format. Will be starting with an empty AddressBook");
-            initialData = new AddressBook();
-        } catch (IOException e) {
-            logger.warning("Problem while reading from the file. Will be starting with an empty AddressBook");
-            initialData = new AddressBook();
-        }
-
-        return new ModelManager(initialData, userPrefs);
+    //Feels like Single responsibility principle is violated if I moved the initialisation phase over to ModelManager?
+    protected Model initModelManager(AlfredStorage alfredStorage, UserPrefs userPrefs) throws AlfredException{
+        return ModelManager.initModelManager(alfredStorage, userPrefs);
```

Comment:
Moved this in my PR to model Manager as well. Dont think SRP is violated anyway. because storage should only be accessed for model ideally

Reply:
ok! 😄